### ISSUE TITLE
Background color state of shape-associated text cannot be changed

### ISSUE THREAD

**Issue by danielbjorkholm** [jgraph/drawio#620]
Hi,

I run 12.1.0 on MacOS, the electron build. Downloaded through website. I have the same issue in the browser (chrome).

I do the following:
1. Pull in shape
2. Double click it to write shape text.
3. Edit the text, b y selecting the shape.
4. Change the text background color, under the text tab, to transparent. 
(or any color really...... or just uncheck the background color altogether.)
5. Select something else, expecting that the background color will change to the selected color.
6. Color remains the same, both on the board and in the right-hand side menu (also remaining checked).

Bonus info:
I also managed to change the text color itself to blue instead of black, which had the effect that the text is now red and the right-hand side menu says that it's blue.

**Comment by jaywarfield** [comment]
Same scenario for IBM VPC where the icon text background color can't be changed from white to transparent so the white background for icon text can stand out too much.  

Further investigation I found that this didn't happen for us in a previous diagram created maybe 9 months ago.  So I can replace a non-working style with a working style to get the background to be transparent.

Working Style:
aspect=fixed;perimeter=ellipsePerimeter;html=1;align=center;shadow=0;dashed=0;image;image=img/lib/ibm/vpc/PublicGateway.svg;fontColor=#000000;labelPosition=center;verticalLabelPosition=bottom;verticalAlign=top;fontSize=12;flipH=1;labelBackgroundColor=none;spacingTop=-5;

Nonworking Style:
aspect=fixed;perimeter=ellipsePerimeter;html=1;align=center;shadow=0;dashed=0;fontColor=#000000;labelBackgroundColor=none;fontSize=12;spacingTop=3;image;image=img/lib/ibm/vpc/PublicGateway.svg;flipH=1;

**Comment by shijianjian** [comment]
Yes

> Same scenario for IBM VPC where we need this fixed as well. The icon text background color can't be changed from white to transparent so the white background for icon text stands out in our new shaded boxes.
> 
> Further investigation for support to consider if what I found is accurate where this didn't happen for us in a previous diagram created maybe 9 months ago. So I can replace a non-working style with a working style to get the background to be transparent. It is interesting to note that both styles below have labelBackgroundColor=none so it looks like the default is suppose to be transparent label background if that is the case.
> 
> Working Style for IBM VPC Public Gateway icon with transparent background for label:
> aspect=fixed;perimeter=ellipsePerimeter;html=1;align=center;shadow=0;dashed=0;image;image=img/lib/ibm/vpc/PublicGateway.svg;fontColor=#000000;labelPosition=center;verticalLabelPosition=bottom;verticalAlign=top;fontSize=12;flipH=1;labelBackgroundColor=none;spacingTop=-5;
> 
> Non-working Style for IBM VPC Public Gateway icon with white background for label that can't be changed:
> aspect=fixed;perimeter=ellipsePerimeter;html=1;align=center;shadow=0;dashed=0;fontColor=#000000;labelBackgroundColor=none;fontSize=12;spacingTop=3;image;image=img/lib/ibm/vpc/PublicGateway.svg;flipH=1;

I found it is interesting that we have to put ```labelBackgroundColor=none``` after the svg image or it will never work.

**Comment by alderg** [comment]
The style contains ...;image;..., which has <add as="labelBackgroundColor" value="#2a2a2a"/>, if you set your own labelBackgroundColor after that, it will override this, if you define it before that, it will be overridden. In general, move entries of the form "class" (ie not key=value) to the start of the style.

**Comment by jaywarfield** [comment]
Thanks for looking into this!  The problem is then with the currently published IBM 1.0 stencils, all of the IBM 1.0 icons have the style set such that labelBackgroundColor is before the image.  I had opened #839 to request that the labelBackgroundColor be moved in the IBM 1.0 icons but I closed #839 at the end of 2020 because we are preparing new icons for IBM 2.0 stencils.  We'll leave IBM 1.0 icons as is then and in the new IBM 2.0 icons the placement will be correct there.  Thanks again!
